### ISSUE TITLE
Brief toolbar icon disappears in transition from 2.3.0rc99 to 2.4.0a2

### ISSUE THREAD

**Issue by thebat137** [ancestorak/brief#259]
Upgrading from 2.3.0rc99 to 2.4.0a2 causes the Brief icon to disappear from my toolbar.  The icon is unavailable in the toolbar customization menu, so I can't restore it.  The icon reappears if I revert Brief to 2.3.0rc99.  I am not sure about intermediate versions, this is just me going between the results of a Firefox automatic upgrade and a version of the .xpi that I had in a backup of my Firefox profile directory.

(I previously accidentally submitted this as a multiprocess compatibility report because the way multiprocess compatibility issue reporting is offered in the addons menu is confusing, so sorry about that...)

**Comment by tanriol** [comment]
Brief 2.4 alphas change the mechanism used for the icon display. It should be available somewhere in the toolbar, or the overflow menu (arrows on toolbar), or the customization menu. If it's not, please try disabling and reenabling Brief. If it still does not appear, please post your OS and Firefox version.

Also you can try the following:
- open the Firefox console (Ctrl+Shift+J) and clear it (trash icon in the console window)
- disable and reenable Brief
- post any errors that appear here

**Comment by thebat137** [comment]
I am using Firefox 52.3.0 under Debian Linux with kernel 4.12.0-1.  Also possibly relevant is that I use the Classic Theme Restorer plugin (version 1.7.0), and I put my Brief icon at the right end of the Additional Toolbar.  It was part of a cluster of several other icons I keep there, and all the others are still there even when the Brief icon disappears.

Upon disabling Brief (version 2.4.0a2) I get the following messages in my console:

```Brief: extension shutdown  bootstrap.js:281
Brief: JSM unload  bootstrap.js:301
Brief: finalized BriefServer
Brief: finalized FeedUpdateService
Bootstrapped manifest not allowed to use 'component' directive.  chrome.manifest:60
Bootstrapped manifest not allowed to use 'contract' directive.  chrome.manifest:61
Bootstrapped manifest not allowed to use 'category' directive.  chrome.manifest:63
Bootstrapped manifest not allowed to use 'overlay' directive.  chrome.manifest:68
Bootstrapped manifest not allowed to use 'overlay' directive.  chrome.manifest:69
Bootstrapped manifest not allowed to use 'style' directive.  chrome.manifest:71
Bootstrapped manifest not allowed to use 'component' directive.  chrome.manifest:74
Bootstrapped manifest not allowed to use 'contract' directive.  chrome.manifest:75
Bootstrapped manifest not allowed to use 'category' directive.  chrome.manifest:76
Warning: attempting to write 4733 bytes to preference extensions.xpiState. This is bad for general performance and memory usage. Such an amount of data should rather be written to an external file. This preference will not be sent to any content processes.
No chrome package registered for chrome://brief/skin/brief-icon-32.png
No matching message handler for the given recipient.  MessageChannel.jsm:621
Use of nsIFile in content process is deprecated.  Content.js:26:17
```

Upon re-enabling it, I get:

```Brief: extension startup  bootstrap.js:239
Warning: attempting to write 4732 bytes to preference extensions.xpiState. This is bad for general performance and memory usage. Such an amount of data should rather be written to an external file. This preference will not be sent to any content processes.
S4E Calling updateWindow: {e9779563-1d50-43e2-af4d-07ca60aeed68}
82  database.js:22:9
Error: Please set webextensions.storage.sync.enabled to true in about:config  (unknown)
A promise chain failed to handle a rejection. Did you forget to '.catch', or did you forget to 'return'?
See https://developer.mozilla.org/Mozilla/JavaScript_code_modules/Promise.jsm/Promise Date: Fri Sep 15 2017 20:30:42 GMT-0400 (EDT)
Full Message: TypeError: Storage is undefined
Full Stack: FeedUpdateServiceInternal_init@resource://brief/FeedUpdateService.jsm:161:9
TaskImpl_run@resource://gre/modules/Task.jsm:319:42
  FeedUpdateService.jsm:161
```

**Comment by tanriol** [comment]
In my case the button did not show up right after update, but showed up after disabling/enabling and shows up after restarting Firefox.

**Comment by tanriol** [comment]
There are some messages that are not related to Brief. As for the related ones:

> Warning: attempting to write 4733 bytes to preference extensions.xpiState. [...]

This may have bad side effects, but nothing like the one we're discussing.

> No chrome package registered for chrome://brief/skin/brief-icon-32.png

Could not reproduce this one yet.

> 82  database.js:22:9

Just debug logging.

> Error: Please set webextensions.storage.sync.enabled to true in about:config

Expected in ESR, harmless.

> TypeError: Storage is undefined

This one (including its explanations) is known and harmless.

**Comment by thebat137** [comment]
Well, okay, that was weird.  I just upgraded to 2.4.0a3 to test your idea about disabling/re-enabling/restarting.  The icon did not appear for me immediately in its normal position, but when I went into the toolbar customization, this time I saw that it had somehow moved from my custom placement (the Additional Toolbar) into my Navigation Bar, which I normally leave disabled.  I am not sure whether it was there on previous tests and I just missed it, but once i found it I was able to drag it back into my Additonal Toolbar and all seems well.  So perhaps the problem is some combination of a) upon update, it is overriding my preferred choice of toolbar for its location and (b) after the update it is not always appearing even in its preferred location until after a restart?

**Comment by tanriol** [comment]
> a) upon update, it is overriding my preferred choice of toolbar for its location

The new button is displayed in a completely different way. Most likely Firefox just did not understand it is the same button and should inherit the previous position.

> (b) after the update it is not always appearing even in its preferred location until after a restart

There was a problem with startup after updates in 2.4.0a2 and below, fixed in 2.4.0a3 with 322a34bdc866b833217649672fc9ff6b56f06f50. It did not look *exactly* like the problem you've described, but I hope the cause was common and fixed now.

Closing this bug as all seems well, feel free to reopen it if the issue reappears. If you have any other issues, you're welcome to report them.
